packages/lit-html/src/lit-html.ts

```diff
@@ -366,17 +371,21 @@ class Template {
             regex = comment2Regex;
           } else if (match[2] !== undefined) {
             if (rawTextElement.test(match[2])) {
-              regex = new RegExp(`<\/${match[2]}`, 'g');
-            } else {
-              regex = tagRegex;
+              // Record if we encounter a raw-text element. We'll switch to
+              // this regex at the end of the tag
+              rawTextTag = new RegExp(`<\/${match[2]}`, 'g');
             }
+            regex = tagRegex;
           } else if (match[3] !== undefined) {
             // dynamic tag name
             regex = tagRegex;
           }
         } else if (regex === tagRegex) {
           if (match[0] === '>') {
-            regex = textRegex;
+            // End of a tag. If we had started a raw-text element, use that
+            // regex
+            regex = rawTextTag ?? textRegex;
+            attrNameEnd = -1;
```

Comment:
Suggest adding a comment about why it's important to reset this here exactly.

Reply:
Added comment